packages/@aws-cdk/aws-codepipeline-actions/lib/bitbucket/source-action.ts

```diff
@@ -98,6 +104,13 @@ export class BitBucketSourceAction extends Action {
     // the action needs to write the output to the pipeline bucket
     options.bucket.grantReadWrite(options.role);
 
+    // if codeBuildCloneOutput is true,
+    // save the connectionArn in the Artifact instance
+    // to be read by the CodeBuildAction later
+    if (this.props.codeBuildCloneOutput === true) {
+      (this.props.output as any)[BitBucketSourceAction.CONNECTION_ARN_PROPERTY] = this.props.connectionArn;
```

Comment:
This is a bit hacky... Why not model it properly into `Artifact` regardless of BitBucket.

Basically you can add `artifact.codeStarConnectionArn` and respect in `CodeBuildAction`.

Reply:
3 reasons:

1. I didn't want to add a public, customer-visible API to `Artifact` that the user is never expected to use.
2. This is something very specific to the `BitBucketSourceAction` and `CodeBuildAction` classes.  I'd rather not add it to `Artifact`, because that would mean this class will accumulate a superset of all features needed by all actions.
3. `BitBucketSourceAction` and `CodeBuildAction` are in the `@aws-cdk/aws-codepipeline-action` package. `Artifact` is in the `@aws-cdk/aws-codepipeline` package. It seems wrong that we would change `Artifact` because a specific action needs something... Imagine `BitBucketSourceAction` lived in a third-party library instead - it would be impossible to change `Artifact` to accommodate it.

That was my thinking process in doing it this way.